users.js

```diff
@@ -272,6 +272,7 @@ class User {
 		this.registered = false;
 		this.userid = toId(this.name);
 		this.group = Config.groupsranking[0];
+		this.lastActive = new Map();
```

Comment:
This is O(nm) memory...

Probably not too much in practice, but is there a need to make it separate for each room?